### PR TITLE
fix: updated reset and help buttons

### LINKED ISSUE #49297: Help Button not Obvious
### Describe the Issue

The new Ask For Help button is not obvious. We have users on the forum frequently not understanding how to find the Ask For Help button.

https://forum.freecodecamp.org/t/new-help-button-not-obvious/571552

### Affected Page

All RWD

### Steps to Reproduce

1. Go to any RWD page
2. Click on 'Check Your Code' a few time
3. See question mark

### Expected behavior

As suggested by @lasjorg, we could include a one word description along with the symbol, for clarity: https://forum.freecodecamp.org/t/new-help-button-not-obvious/571552/7

### Screenshots

_No response_

### System

All

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by DEVelooper29**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #49297

### CODE REVIEW

**Comment by DEVelooper29** [comment]
Screenshot (2880)
Screenshot (2881)
Thank you for giving me the chance to contribute. KIndly review and please let me know if any changes/suggestions.

**Review by Sboonny — CHANGES_REQUESTED** [review]
- Please revert the changes in the SVG, the SVGs are in button so you can add the text to the button instead.
- please don't use `width` and `height`, they create rigid container.
- the icon-bar won't only have reset-help, a share button is being added, so the new name is more harmful than helpful.
- you shouldn't use the media query, you can change the `reset-help-icon-bar` display to grid, and make it shift its children layout.
- the strings should be translatable, there are examples in the lowerjaw button of how you can make them so.

**Comment by DEVelooper29** [comment]
Thank you for your suggestions. I'll try my best to include all of them.

**Comment by DEVelooper29** [comment]
> * Please revert the changes in the SVG, the SVGs are in button so you can add the text to the button instead.
> * please don't use `width` and `height`, they create rigid container.
> * the icon-bar won't only have reset-help, a share button is being added, so the new name is more harmful than helpful.
> * you shouldn't use the media query, you can change the `reset-help-icon-bar` display to grid, and make it shift its children layout.
> * the strings should be translatable, there are examples in the lowerjaw button of how you can make them so.

Sir, the first change you suggested can only be achieved using media queries. And in the changes requested by Sboony I've been told not to use media queries. So, what should I do?

**Comment by Sboonny** [comment]
Responsiveness can be achieved without media query, when you're done with the other suggestions, mention this in discord and I or another one will guide you through it

**Comment by DEVelooper29** [comment]
The suggested changes done:
1) Please use responsive units as much as possible.
2) There UI is also served in RTL, so the UI elements should look mirrored and presentable in that state.
3) Move styles in a CSS file.
4) The panes can move independently from the view size, so changing the styles based on screen width will not work
5) Please revert the changes in the SVG, the SVGs are in button so you can add the text to the button instead.
6) the icon-bar won't only have reset-help, a share button is being added, so the new name is more harmful than helpful.
7) you shouldn't use the media query, you can change the reset-help-icon-bar display to grid, and make it shift its children layout.
8) the strings should be translatable, there are examples in the lowerjaw button of how you can make them so.
9) Since the buttons will now have actual text inside of them we need to remove the aria-label attributes from the them. Screen reader users should hear the exact text in the button and no more, so there is no need to use aria-label as the screen reader will automatically read the text in the button.
10) I would strongly urge we remove the title attribute from the buttons. Now that we have text for the buttons there really is no need to have a tooltip. If we truly feel that we still need a tooltip because the button text is not clear enough then we need to re-evaluate the whole thing.

Suggested changes almost done:
1) please don't use width and height, they create rigid container. (I've tried to use height width as minimum as possible and tried to use responsive units for it)

Suggested changes where I am stuck a bit:
1) Make sure the text is centered and buttons are full width when they are in a column.
2) Different translations have different word length. The style changes should account for that.

File: client/src/assets/icons/help.css
Comment (bbsmooth) [inline client/src/assets/icons/help.css:12]: These can't be in `vh`, they need to be in a proportional unit like `rem`. Otherwise, as you zoom in they will not grow accordingly. This is the Help button at 400% zoom:
zoomed-help-button

File: client/src/assets/icons/help.css
Comment (bbsmooth) [inline client/src/assets/icons/help.css:20]: Same deal here, need to be in a proportional unit to allow for zoom.

Comment (DEVelooper29) [reply client/src/assets/icons/help.css:20]: got it.

File: client/src/templates/Challenges/classic/lower-jaw.css
Comment (bbsmooth) [inline client/src/templates/Challenges/classic/lower-jaw.css:10]: I would urge you to reconsider absolute positioning here. I think it might be much easier to use flex on the button itself to position the text and icon. Also, this will fix your problems with rtl direction.

File: client/src/templates/Challenges/classic/editor.css
Comment (Sboonny) [inline client/src/templates/Challenges/classic/editor.css:183]: To answer the question on discord you need to use `minmax`, to make it take full width, you will need to manipulate the `5em` value tho, to get a good looking button, and remove alot of properties


```suggestion
  grid-template-columns: repeat(auto-fill, minmax(5em,1fr));
```

**Review by Sboonny — COMMENTED** [review]
@DEVelooper29, please delete the new files and the added classNames to the SVG, there are 3 files that should be changed `editor.css`, `lower-jaw.tsx`, and `translation.json`

**Comment by DEVelooper29** [comment]
> @DEVelooper29, please delete the new files and the added classNames to the SVG, there are 3 files that should be changed `editor.css`, `lower-jaw.tsx`, and `translation.json`

Okay sir I'll do that. Actually I made new css files because I had got a suggestion to do that. But I didn't get what should be changed in 'translations.json' as you mentioned now. Because it already has reset and help messages and I have already used them in my latest push.

**Review by Iamali0 — COMMENTED** [review]
> DEVelooper29:help_reset`

**Comment by bbsmooth** [comment]
@DEVelooper29 This is looking better but I still think there are styling issues. Are you testing by adjusting the width of the editor iframe? How about zooming in the page up to 400%? As I am making these changes I am noticing that the widths of the buttons are changing out of proportion to their content and the spacing between the icon and the text is changing. I could be wrong, maybe that's what we want? I guess we'll need some clarification. But I'm thinking that these things should be consistent as you change the width of the iframe or zoom in.

**Comment by DEVelooper29** [comment]
> @DEVelooper29 This is looking better but I still think there are styling issues. Are you testing by adjusting the width of the editor iframe? How about zooming in the page up to 400%? As I am making these changes I am noticing that the widths of the buttons are changing out of proportion to their content and the spacing between the icon and the text is changing. I could be wrong, maybe that's what we want? I guess we'll need some clarification. But I'm thinking that these things should be consistent as you change the width of the iframe or zoom in.

First of all, Thank you so much for your review. I've zoomed the page but I guess it didn't affect the stylings. But yeah, width of the buttons and spacing between svgs and text is changing as the size of page changes. But that's because of the code suggested by a member so that the width of buttons get full when they are arranged in columns. So because of this line the width of buttons and spacing between text and svg is changing. I've tried my best to include all the suggestions of the reviewers. But still if anything else is to be done, please let me know what it is and its approach.

**Review by Sboonny — COMMENTED** [review]
> Different translations have different word length. The style changes should account for that.

You can find the translations in translations.json, to save time the longest word is  `Zurücksetzen`, you can add the word as hard value in the reset button for now as draft for your change and after making the necessary changes revert it.

For the help, the longest word is `Допомога`

File: client/src/assets/icons/reset.tsx
Comment (DEVelooper29) [reply client/src/assets/icons/reset.tsx:24]: But if I don't add wrapper and div, how will I style the wrapper?

Comment (Sboonny) [reply client/src/assets/icons/reset.tsx:24]: You won't be styling the wrapper in this file, you will delete it as well.

if you are asking for why, because you are adding `flex` to `lower-utility-bar > button` so this SVG is already in a flex container and has `align-items`.

**Comment by Sboonny** [comment]
@DEVelooper29, is there a update for this PR?

**Comment by bbsmooth** [comment]
Functionally, this is fine. But stylistically, I think it still needs work. Is it OK if I take a crack at this?

**Comment by Sboonny** [comment]
Go for it 👍, I will assign you, when you think it's ready for review, unassign yourself.

**Comment by bbsmooth** [comment]
I think for handling other languages we will need to set a custom minimum width on the `.utility-bar` grid columns based on the longest word for each language (in client/src/templates/Challenges/classic/editor.css).

**Comment by Sboonny** [comment]
I will defer to ahmad about the multiple language style.

**Review by moT01 — APPROVED** [review]
I would be fine getting this in if you all are done with it.

**Comment by bbsmooth** [comment]
I've added the button width customizations for the various languages we support. The two Chinese languages can use the default (English) width since their words for "Reset" and "Help" are not wider than their English counterparts. All the other languages needed to be adjusted as they had at least one word that was wider than its English counterpart.

**Comment by moT01** [comment]
Isn't there a way to make this more dynamic? Something that would work on all languages without setting each language. I don't mind it here since it's the only place we've done this that I know of - but it feels like a maintenance pain. We're going to add a new language and forget to set this.

**Comment by bbsmooth** [comment]
> Isn't there a way to make this more dynamic?

Sure, but I think we might need to violate @Sboonny 's rule above about not using media queries if we want to avoid using JS. 

I'll make some changes and we can see how everyone feels about them.

**Comment by bbsmooth** [comment]
I thought using container queries (in browsers that support it) would gain us some advantage but it really doesn't. I think the best way to do this is to set a min width on the buttons that will accommodate the largest word possible from all of the languages. So ya, when we add another language we will need to adjust the min width if that language has a longer word for either Reset or Help than German's word for Reset ( Zurücksetzen). 

That is, if we want to keep this a purely CSS solution. I'm guessing we could get JS involved to automatically figure out the min width for the buttons.

**Comment by Sboonny** [comment]
> Sure, but I think we might need to violate @Sboonny 's rule above about not using media queries if we want to avoid using JS.

if it's possible to have the button shift when the container change `width` similar to what happens when the viewport does, go for it.

---

How do we account for the word "share", which isn't translated, or if someone decided a longer different word is a better translation for the current button context.

I think it's the best to use:

- `auto` in the grid and not mind the buttons having different widths.
- JavaScript to set them the width.

**Comment by bbsmooth** [comment]
> * `auto` in the grid and not mind the buttons having different widths.

Ahh, I thought having the buttons all the same width was a requirement. This changes things a little :slightly_smiling_face:

**Comment by bbsmooth** [comment]
> How do we account for the word "share", which isn't translated

Where is the word "share" being used?

**Comment by Sboonny** [comment]
the share button will be added in this PR https://github.com/freeCodeCamp/freeCodeCamp/pull/49395.

> What if all the buttons cannot fit side by side on one line? Can they line wrap and just the last button moves to the next line? Or do they need to switch to a single column and the buttons take up the entire width of the lower jaw?

These are good questions that I haven't thought of 🤔, we will be sacrificing the `auto-fill`, which means that all the question can be answered by "nope".

I will take a deeper look at when I comeback on Monday, but this PR shouldn't wait for my feedback feel free to add JS and I will rework on this later.

**Review by Sboonny — APPROVED** [review]
other than this nitpick, I happy with merging this

File: client/src/templates/Challenges/classic/editor.css
Comment (Sboonny) [inline client/src/templates/Challenges/classic/editor.css:183]: The nitpick:

The grid override these values, so they don't shift, unless I have missed something.

```suggestion
```

**Comment by bbsmooth** [comment]
@Sboonny I've removed those unnecessary CSS properties. 

Ultimately, if we are not 100% happy with how this works then I think we need to open a new issue and discuss the fine details over how these buttons should behave, because if feels like we have too many unanswered questions right now to adjust this PR any further.

**Pull request merged**
